Rubberduck.CodeAnalysis/Inspections/Abstract/DeclarationInspectionBaseBase.cs

```diff
@@ -13,15 +13,15 @@ public abstract class DeclarationInspectionBaseBase : InspectionBase
         protected readonly DeclarationType[] RelevantDeclarationTypes;
         protected readonly DeclarationType[] ExcludeDeclarationTypes;
 
-        protected DeclarationInspectionBaseBase(RubberduckParserState state, params DeclarationType[] relevantDeclarationTypes)
-            : base(state)
+        protected DeclarationInspectionBaseBase(IDeclarationFinderProvider declarationFinderProvider, params DeclarationType[] relevantDeclarationTypes)
```

Comment:
This is why I'm less certain than before about the *one class: one file* rule/guideline, @bclothier: I'm thinking the only reason this class needs a silly name here is *because* of *one class: one file*, because `DeclarationInspectionBase<T>` wants to go into a `DeclarationInspectionBase.cs` file.
I'm pretty sure we could have `DeclarationInspectionBase` and `DeclarationInspectionBase<T>` in the same `DeclarationInspectionBase.cs` file, and avoid needing to name our abstract classes `ThingBaseBase` 😜 

@MDoerner what do you think?

Reply:
I'll admit that I am not consistent on that guideline myself. But FWIW, in this particular case, I had files [`RefactoringDialogBase`](https://github.com/rubberduck-vba/Rubberduck/blob/next/Rubberduck.Core/UI/Refactorings/RefactoringDialogBase.cs) and [`RefactoringDialogBaseGeneric`](https://github.com/rubberduck-vba/Rubberduck/blob/next/Rubberduck.Core/UI/Refactorings/RefactoringDialogBaseGeneric.cs). Note that in the generic file, the class is just `RefactoringDialogBase<>`; no need for `BaseBase`.

Reply:
Ah, but then you sacrifice the consistency of `classname.cs`! #NamingIsHard

Reply:
Also, in this particular example, this isn't generic. It's basically a `InspectionBase` implementing a particular implementation for declaration inspections. In which case, I'm not sure it should be `DeclarationInspectionBaseBase`; simply just `DeclarationInspectionBase` is sufficient. The fact that it's based on another abstract class (`InspectionBase`) is not really material and shouldn't factor into the naming, IMO.

Yes, agreed on #NamingIsHard.

Reply:
I have to admit that I don't really like the name myself. It is called that way because it is the base class for the other declaration inspection base classes. Apart from the ones you already mentioned there are also the two multi result base classes.
That it is based on `InspectionBase` is irrelevant for the name.

Reply:
Btw, the generic and non-generic versions of the base classes are already in the same file for each kind.

Reply:
Should I do something about this class or not? Options are

1. Rename `DeclarationInspectionBase` to `DeclaratiinInspectionSingleResultBase` and `DeclarationInspectionBaseBase` to `DeclarationInspectionBase`. That removes the `BaseBase` but opens room for confusion since this class should exclusively be implemented by base classes and not concrete inspection.
2. Inline the class into all three implementation inspection base classes. 
3. Leave it as-is.

Reply:
Your comment makes me realize that I had the `BaseBase` backward; I interpreted it as a `Base` class based on another `Base` class but you're meaning a `Base` class that needs to be implemented by other `Base` classes. I had to look at the `DeclarationInspectionBase` to see that it inherits `DeclarationInspectionBaseBase`. Obviously that ambiguity is not what we want. 

The only suggestion I can think of is `PartialBase`, in similar fashion to the `partial` class to suggest that it's not a complete definition, but I am not sure that is any improvement. At very least, there should be a xml docs on the `BaseBase` explaining that it is meant to be implemented by other base classes and not directly.